### PR TITLE
issue-823-i18n-active-passive-scan-rules-beta

### PR DESCRIPTION
**Pull request by kingthorin**
Fixes zaproxy/zaproxy#823

### CODE REVIEW

File: src/org/zaproxy/zap/extension/ascanrulesBeta/ExpressionLanguageInjectionPlugin.java [hunk 1]
```diff
@@ -184,7 +184,7 @@ public void scan(HttpMessage msg, String paramName, String value) {
                 this.bingo(
                         Alert.RISK_HIGH,
                         Alert.CONFIDENCE_MEDIUM,
-                        null,
+                        getName(),
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/ascanrulesBeta/ExpressionLanguageInjectionPlugin.java]: The 3rd parameter is the URI (in this bingo method).

File: src/org/zaproxy/zap/extension/ascanrulesBeta/PaddingOraclePlugin.java [hunk 2]
```diff
@@ -216,7 +216,7 @@ private boolean checkPaddingOracle(String paramName, String value, OracleEncoder
                             this.bingo(
                                     Alert.RISK_HIGH,
                                     Alert.CONFIDENCE_MEDIUM,
-                                    null,
+                                    getName(),
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/ascanrulesBeta/PaddingOraclePlugin.java]: The 3rd parameter is the URI (in this bingo method).

File: src/org/zaproxy/zap/extension/ascanrulesBeta/PaddingOraclePlugin.java [hunk 3]
```diff
@@ -243,7 +243,7 @@ private boolean checkPaddingOracle(String paramName, String value, OracleEncoder
...
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/ascanrulesBeta/PaddingOraclePlugin.java]: Ditto.

File: src/org/zaproxy/zap/extension/ascanrulesBeta/ZapAddOn.xml [hunk 4]
```diff
@@ -1,12 +1,12 @@
 <zapaddon>
 	<name>Active scanner rules (beta)</name>
-	<version>19</version>
+	<version>20</version>
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/ascanrulesBeta/ZapAddOn.xml]: The new version (19) of the add-on was not yet released.

File: src/org/zaproxy/zap/extension/ascanrulesBeta/ZapAddOn.xml [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
 	<description>The beta quality Active Scanner rules</description>
 	<author>ZAP Dev Team</author>
 	<url/>
 	<changes>
 	<![CDATA[
-	Adding Integer Overflow Scanner.<br>
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/ascanrulesBeta/ZapAddOn.xml]: This change should be kept (as per previous comment).

File: src/org/zaproxy/zap/extension/pscanrulesBeta/ViewstateScanner.java [hunk 6]
```diff
@@ -458,15 +465,15 @@ private void setVersion() {
         
         /* TODO once we have good Viewstate 1 & 2 parsers */
         public Object[] getObjectTree() throws Exception {
-        	throw new Exception("Not implemented (yet)");
+        	throw new Exception(Constant.messages.getString(MESSAGE_PREFIX + "exception"));
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/pscanrulesBeta/ViewstateScanner.java]: The exception message should not be internationalised, it's not shown to the user.
Same comment for following exceptions.

File: src/org/zaproxy/zap/extension/pscanrulesBeta/ZapAddOn.xml [hunk 7]
```diff
@@ -1,12 +1,12 @@
 <zapaddon>
 	<name>Passive scanner rules (beta)</name>
-	<version>13</version>
+	<version>14</version>
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/pscanrulesBeta/ZapAddOn.xml]: The new version (13) of the add-on was not yet released.

File: src/org/zaproxy/zap/extension/pscanrulesBeta/ZapAddOn.xml [hunk 8]
```diff
@@ -1,12 +1,12 @@
...
 	<description>The beta quality Passive Scanner rules</description>
 	<author>ZAP Dev Team</author>
 	<url/>
 	<changes>
 	<![CDATA[
-	Issue 1966: Charset Mismatch passive scanner now handles HTML5 meta charset, and multiple conditions.<br>
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/pscanrulesBeta/ZapAddOn.xml]: This change should be kept (as per previous comment).

File: src/org/zaproxy/zap/extension/pscanrulesBeta/resources/Messages.properties [hunk 9]
```diff
@@ -33,6 +33,30 @@ pscanbeta.cookielooselyscoped.refs=http://code.google.com/p/browsersec/wiki/Part
 pscanbeta.cookielooselyscoped.extrainfo=The origin domain used for comparison was: \r\n{0}\r\n{1}
 pscanbeta.cookielooselyscoped.extrainfo.cookie={0}\r\n
 
+pscanbeta.informationdisclosuredebugerrors.name=Information Disclosure - Debug Error Messages
+pscanbeta.informationdisclosuredebugerrors.desc=The response appeared to contain common error messages returned by platforms such as ASP.NET, and Web-servers such as IIS and Apache. You can configure the list of common debug messages.
+pscanbeta.informationdisclosuredebugerrors.soln=Disable debugging messages before pushing to production.
+
+pscanbeta.informationdisclosureinurl.name=Information Disclosure - Sensitive Informations in URL
+pscanbeta.informationdisclosureinurl.desc=The request appeared to contain sensitive information leaked in the URL. This can violate PCI and most organizational compliance policies. You can configure the list of strings for this check to add or remove values specific to your environment.
+pscanbeta.informationdisclosureinurl.otherinfo.senitiveinfo=The URL contains potentially sensitive information.
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/pscanrulesBeta/resources/Messages.properties]: There's a typo in the resource message key ("se**ni**tiveinfo").

File: src/org/zaproxy/zap/extension/pscanrulesBeta/resources/Messages.properties [hunk 10]
```diff
@@ -33,6 +33,30 @@ pscanbeta.cookielooselyscoped.refs=http://code.google.com/p/browsersec/wiki/Part
...
+pscanbeta.informationdisclosureinurl.otherinfo.cc=The URL appears to contain credit card information.
+pscanbeta.informationdisclosureinurl.otherinfo.email=The URL contains email address(es).
+pscanbeta.informationdisclosureinurl.otherinfo.ssn=The URL appears to contain US Social Security Number(s)
+pscanbeta.informationdisclosureinurl.soln=Do not pass sensitive information in URIs.
+
+pscanbeta.informationdisclosurereferrerscanner.name=Information Disclosure - Sensitive Information in HTTP Referrer Header
+pscanbeta.informationdisclosurereferrerscanner.desc=The HTTP header may have leaked a potentially sensitive parameter to another domain. This can violate PCI and most organizational compliance policies. You can configure the list of strings for this check to add or remove values specific to your environment.
+pscanbeta.informationdisclosurereferrerscanner.otherinfo.senitiveinfo=The URL in the HTTP referrer header field appears to contain sensitive information.
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/pscanrulesBeta/resources/Messages.properties]: There's a typo in the resource message key ("se**ni**tiveinfo").

**Comment by thc202** [comment]
The Csrftokenscan's name and the scanner ServletParameterPollutionScanner are not internationalised.
Do you plan to do that? If so, in this PR?

**Comment by kingthorin** [comment]
Thanks @thc202, great catches. I've just addressed all these, I'll get the other two you mentioned done tonight and submit a fixup.

File: src/org/zaproxy/zap/extension/pscanrulesBeta/resources/Messages.properties [hunk 11]
```diff
@@ -47,4 +71,32 @@ pscanbeta.insecurejsfviewstate.extrainfo=JSF ViewState [{0}] is insecure
 pscanbeta.noanticsrftokens.name=Absence of Anti-CSRF Tokens
 pscanbeta.noanticsrftokens.desc=No Anti-CSRF tokens were found in a HTML submission form.
 pscanbeta.noanticsrftokens.alert.attack=None. Warning only.
-pscanbeta.noanticsrftokens.alert.extrainfo=No known Anti-CSRF tokens {0} were found in the following HTML forms: {1}.  
\ No newline at end of file
+pscanbeta.noanticsrftokens.alert.extrainfo=No known Anti-CSRF tokens {0} were found in the following HTML forms: {1}.
+
+pscanbeta.servletparameterpollutionscanner.name=HTTP Parameter Override
+pscanbeta.servletparameterpollutionscanner.desc=Unspecified form action: HTTP parameter override attack potentially possible. This is a known problem with Java Servlets but other platforms may also be vulnerable."
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/pscanrulesBeta/resources/Messages.properties]: There's a quote char at the end of the line.

**Comment by thc202** [comment]
The branch should to be rebased (because of the passive ZapAddOn.xml file).

**Comment by kingthorin** [comment]
I've got this branch in a completely messed up state now....I hate git.

I think I'm going to have to copy all the content. Remove this branch and start all over again.

I'm stuck in either "Rebase w/ Merge" mode or in a state where Commit/Push doesn't actually do anything though it shows as completing properly.

**Comment by thc202** [comment]
Thanks!

**Pull request merged**